bundles/org.openhab.binding.souliss/src/main/java/org/openhab/binding/souliss/handler/SoulissT11Handler.java

```diff
@@ -0,0 +1,158 @@
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+=======
+ * Copyright (c) 2014-2019 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.souliss.handler;
+
+import java.math.BigDecimal;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.souliss.SoulissBindingConstants;
+import org.openhab.binding.souliss.SoulissBindingProtocolConstants;
+import org.openhab.core.config.core.Configuration;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.PrimitiveType;
+import org.openhab.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SoulissT11Handler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Tonino Fazio - Initial contribution
+ * @author Luca Calcaterra - Refactor for OH3
+ */
+
+@NonNullByDefault
+public class SoulissT11Handler extends SoulissGenericHandler {
+
+    private @NonNullByDefault({}) Configuration gwConfigurationMap;
+    private Logger logger = LoggerFactory.getLogger(SoulissT11Handler.class);
+    byte t1nRawState;
+    byte xSleepTime = 0;
+
+    public SoulissT11Handler(Thing thing) {
+        super(thing);
+    }
+
+    // called on every status change or change request
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            switch (channelUID.getId()) {
+                case SoulissBindingConstants.ONOFF_CHANNEL:
+                    @Nullable
+                    OnOffType val = getOhStateOnOffFromSoulissVal(t1nRawState);
+                    if (val != null) {
+                        updateState(channelUID, val);
+                    }
+                    break;
+                default:
+                    logger.debug("Unknown channel for T11 thing: {}", channelUID);
+            }
+        } else {
+            switch (channelUID.getId()) {
+                case SoulissBindingConstants.ONOFF_CHANNEL:
+                    if (command instanceof OnOffType) {
+                        if (command.equals(OnOffType.ON)) {
+                            commandSEND(SoulissBindingProtocolConstants.SOULISS_T1N_ON_CMD);
+                        } else if (command.equals(OnOffType.OFF)) {
+                            commandSEND(SoulissBindingProtocolConstants.SOULISS_T1N_OFF_CMD);
+                        }
+                    }
```

Comment:
```suggestion
                        if (command.equals(OnOffType.ON)) {
                            commandSEND(SoulissBindingProtocolConstants.SOULISS_T1N_ON_CMD);
                        } else if (command.equals(OnOffType.OFF)) {
                            commandSEND(SoulissBindingProtocolConstants.SOULISS_T1N_OFF_CMD);
                        }
```

Uses less cpu to do same thing.

Reply:
fixed